purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesOrchestrator.kt

```diff
@@ -1128,13 +1158,36 @@ internal class PurchasesOrchestrator(
         isPersonalizedPrice: Boolean?,
         purchaseCallback: PurchaseCallback,
     ) {
+        trackPurchaseStarted(purchasingData.productId, purchasingData.productType)
+        val startTime = dateProvider.now
+
+        val callbackWithDiagnostics = object : PurchaseCallback {
+            override fun onCompleted(storeTransaction: StoreTransaction, customerInfo: CustomerInfo) {
+                trackPurchaseResult(
+                    purchasingData,
+                    error = null,
+                    startTime,
+                    customerInfo.entitlements.verification,
+                )
+                purchaseCallback.onCompleted(storeTransaction, customerInfo)
+            }
+
+            override fun onError(error: PurchasesError, userCancelled: Boolean) {
+                trackPurchaseResult(
+                    purchasingData,
+                    error,
+                    startTime,
+                    verificationResult = null,
+                )
+                purchaseCallback.onError(error, userCancelled)
+            }
+        }
+
         if (purchasingData.productType != ProductType.SUBS) {
-            purchaseCallback.dispatch(
-                PurchasesError(
-                    PurchasesErrorCode.PurchaseNotAllowedError,
-                    PurchaseStrings.UPGRADING_INVALID_TYPE,
-                ).also { errorLog(it) },
-            )
+            PurchasesError(
+                PurchasesErrorCode.PurchaseNotAllowedError,
+                PurchaseStrings.UPGRADING_INVALID_TYPE,
+            ).also { errorLog(it) }.also { callbackWithDiagnostics.dispatch(it) }
```

Comment:
Doing this to avoid the linter's error due to `LongMethod`. I needed one less line 😄